File: ordercoffeeshopfrontend/src/services/shippingService.ts
import api from './api';

export interface ShippingInfo {
  shippingFee: number;
  distance: number;
  storeId: number;
  storeName: string;
}

export const calculateShippingFeeForStore = async (storeId: number, latitude: number, longitude: number): Promise<ShippingInfo> => {
  const response = await api.post('/shipping/calculate-for-store', { storeId, latitude, longitude });
  return response.data;
};

export const calculateShippingFee = async (latitude: number, longitude: number): Promise<ShippingInfo> => {
  const response = await api.post('/shipping/calculate', { latitude, longitude });
  return response.data;
};

File: ordercoffeeshopbackend/src/main/java/com/coffeeshop/backend/service/ShippingService.java
package com.coffeeshop.backend.service;


import com.coffeeshop.backend.dto.ShippingInfoDTO;

public interface ShippingService {
    ShippingInfoDTO calculateShippingFee(double latitude, double longitude);
    ShippingInfoDTO calculateShippingFeeForStore(Long storeId, double latitude, double longitude);
}
